### PR TITLE
daterange bug fix for manually entered dates

### PR DESCRIPTION
**Pull request by stephherbers**
## Product Description

Ticket here: https://dimagi-dev.atlassian.net/browse/USH-1328
The value would clear after copying in or typing in a date with format like ‘2000/8/8’

## Technical Summary

The date would clear even with formats like "2000-8-8" as well ,so the issue actually lied in it being a single date instead of a range using the separator. This PR adds a condition if the input is a single valid date is typed in rather than a range. 

### Safety story

locally tested and ensured the dates that previously work still work after this change

### QA Plan

I am not requesting QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js [hunk 1]
```diff
@@ -249,6 +249,10 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                     oldValue = $input.val(),
                     parts = _.map(oldValue.split(separator), function (v) { return moment(v); }),
                     newValue = '';
+                // conditions where only one valid date is typed in rather than a range
+                if (parts.length === 1 && function (parts) { return parts.isValid(); }) {
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js:253]:
```suggestion
                if (parts.length === 1 && function (parts) { return parts.isValid(); }()) {
```

**Review by orangejenny — CHANGES_REQUESTED** [review]
The second part of the condition will be true even if the date isn't valid, because the inline function doesn't get called.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js (hunk 1 above)
Comment (stephherbers) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js]: ope thank you-- that brought up too that parts should be checked the same as the other conditional so I modified to check if the parts are a valid length before the length check

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js [hunk 2]
```diff
@@ -249,8 +249,14 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                     oldValue = $input.val(),
                     parts = _.map(oldValue.split(separator), function (v) { return moment(v); }),
                     newValue = '';
-                if (parts.length === 2 && _.every(parts, function (part) { return part.isValid(); })) {
-                    newValue = parts[0].format(dateFormat) + separator + parts[1].format(dateFormat);
+
+                if (_.every(parts, function (part) { return part.isValid(); }))  {
+                    if (parts.length === 1) { // condition where only one valid date is typed in rather than a range
+                        $input.val(oldValue + separator + oldValue).trigger('change');
+                    }
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js:256]: our js uses [cuddled elses](https://twitter.com/chaelcodes/status/1362542890027790336?lang=en)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js [hunk 3]
```diff
@@ -249,8 +249,14 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
...
+                    else if (parts.length === 2) {
+                        newValue = parts[0].format(dateFormat) + separator + parts[1].format(dateFormat);
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js:258]: it looks like there's an error above, a semicolon that should be a comma so `dateFormat` doesn't get declared with `var`: https://github.com/dimagi/commcare-hq/blob/0a0c3d9e15d04728f52a0c2c69050c6832b3e605/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js#L8

**Review by orangejenny — APPROVED** [review]

**Pull request merged**